test/PHPUnit/Controller/TemplateNameTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Test\PHPUnit\Controller;
+
+use LaminasTest\Test\PHPUnit\Controller\AbstractHttpControllerTestCaseTest;
+
+class TemplateNameTest extends AbstractHttpControllerTestCaseTest
```

Comment:
It'a a bit weird that you extended this test in a new class - it means that all the inherited test methods will run _again_ - I would expect a new test _method_ added to the test case around here: https://github.com/laminas/laminas-test/blob/31a9a0c9e7a6ff1d422f850dbbbb7dd664892418/test/PHPUnit/Controller/AbstractControllerTestCaseTest.php#L526
… with a method name something like `testSearchTemplatesVerifiesDeeplyNestedTemplateName`.

The content of the test method seems fine to me though 👍

Reply:
This should go into abstract controller test adjacent to the template assertion test or fixing template assertion test.

Reply:
Yes. I was debating this myself.
I wanted to stress that I was testing a specific case raised by the issue to make it clear.
I can move the test case as a method in `AbstractControllerTestCaseTest.php`.

Reply:
Done.
I also updated the commit messages.